pkg/server/endpoints/authorized_entryfetcher_attested_nodes_test.go

```diff
@@ -9,62 +9,83 @@ import (
 	"github.com/sirupsen/logrus"
 	"github.com/sirupsen/logrus/hooks/test"
 	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
 	"github.com/spiffe/spire/pkg/server/authorizedentries"
 	"github.com/spiffe/spire/pkg/server/datastore"
 	"github.com/spiffe/spire/proto/spire/common"
 	"github.com/spiffe/spire/test/clock"
 	"github.com/spiffe/spire/test/fakes/fakedatastore"
+	"github.com/spiffe/spire/test/fakes/fakemetrics"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestUpdateAttestedNodesCache(t *testing.T) {
-	ctx := context.Background()
-	log, _ := test.NewNullLogger()
-	clk := clock.NewMock(t)
-	ds := fakedatastore.New(t)
-	cache := authorizedentries.NewCache(clk)
-
-	attestedNodes, err := buildAttestedNodesCache(ctx, log, ds, clk, cache, defaultSQLTransactionTimeout)
-	require.NoError(t, err)
-	require.NotNil(t, attestedNodes)
-
-	agentID, err := spiffeid.FromString("spiffe://example.org/myagent")
-	require.NoError(t, err)
-
-	_, err = ds.CreateAttestedNode(ctx, &common.AttestedNode{
-		SpiffeId:     agentID.String(),
-		CertNotAfter: time.Now().Add(5 * time.Hour).Unix(),
-	})
-	require.NoError(t, err)
-
 	for _, tt := range []struct {
 		name                            string
 		errs                            []error
 		expectedLastAttestedNodeEventID uint
+		expectMetrics                   []fakemetrics.MetricItem
 	}{
 		{
 			name:                            "Error Listing Attested Node Events",
 			errs:                            []error{errors.New("listing attested node events")},
 			expectedLastAttestedNodeEventID: uint(0),
+			expectMetrics:                   nil,
 		},
 		{
 			name:                            "Error Fetching Attested Node",
 			errs:                            []error{nil, errors.New("fetching attested node")},
 			expectedLastAttestedNodeEventID: uint(0),
+			expectMetrics:                   nil,
 		},
 		{
 			name:                            "Error Getting Node Selectors",
 			errs:                            []error{nil, nil, errors.New("getting node selectors")},
 			expectedLastAttestedNodeEventID: uint(0),
+			expectMetrics:                   nil,
 		},
 		{
 			name:                            "No Errors",
 			expectedLastAttestedNodeEventID: uint(1),
+			expectMetrics: []fakemetrics.MetricItem{
+				{
+					Type:   fakemetrics.SetGaugeType,
+					Key:    []string{telemetry.Node, telemetry.AgentsByExpiresAtCache, telemetry.Count},
+					Val:    1,
+					Labels: nil,
+				},
+				{
+					Type:   fakemetrics.SetGaugeType,
+					Key:    []string{telemetry.Node, telemetry.AgentsByIDCache, telemetry.Count},
+					Val:    1,
+					Labels: nil,
+				},
+			},
 		},
 	} {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
```

Comment:
Good moving the setup inside `t.Run` to avoid sharing setup between test cases in the table. Perhaps consider extracting to a helper?

It looks like this is the only table where you did this move in the endpoints tests touched. Was this the only test where the changes uncovered an issue with shared setup? Should we be consistent and move all setup inside `t.Run`?

Reply:
Yeah, this was the only test that this was an issue in.